### DISCUSSION TITLE
Support for Azure Synapse

### DISCUSSION THREAD

**Discussion by shireyjw** [ScalefreeCOM/datavault4dbt#60] [Ideas]
What would it take to support Azure Synapse?  Is it on the roadmap?

**Comment by tkirschke** [comment]
Hi @shireyjw and thanks for reaching out!

Support for Azure Synapse is definetely on the roadmap! Out current timeline aims for a release of the adapter latest within this year, but of course we try everything to allocate resources for development ASAP.

Nevertheless, we would be happy to receive work from the community here! So if you can't wait to use Datavault4dbt on Azure Synapse, then feel free to start working on the adapter for yourself! And whenever you have something that works breifly, we are happy to review a Pull Request and to include you solution into the publicly available package! 

## Short tips on how to start building an adapter: 

- Set up a demo environment of your desired target database, or use existing
- Set up dbt project (or use existing) that is connected with the target database
- Install Datavault4dbt
- Copy all Datavault4dbt macros of one existing database adapter into a new folder inside your own macros folder - we recommend to copy from either the BigQuery or Snowflake macros
  - Typically, you would only have to copy over the macros within the "staging" and "tables" folder of Datavault4dbt. All other background macros are usually more flexible and shouldn't need to be modified by default
- Rename all these macros and remove the "default__" (for BigQuery) or "Snowflake__" (for Snowflake) prefix of the macro names. Replace it by the name of your database adapter
- Now you can start applying those macros to demo models. Typically you would start building a staging model and therefore apply the staging macro, then continue with all other macros
  - For all macros and their testing models:
    - Execute model
    - Copy the failing code (if failing) out of the target/run folder into a database query input
    - Modify the SQL to make it work on the target database
    - Adjust the macros of your database adapter to generate the working SQL
